### PR TITLE
Upgrade sigstore/cosign dependency to v1.5.2

### LINKED ISSUE #923: Upgrade sigstore/cosign dependency version
Currently, the imported version of sigstore/cosign in kpack is v1.0.0. However, version v1.5.1 is now available and it would be good to upgrade this dependency prior to implementing signing for `Builders`/`ClusterBuilders`.

### PR DESCRIPTION
**Pull request by stormqueen1990**
Upgrade sigstore/cosign dependency from v1.0.0 to v1.5.2.

Fixes #923 

Tasks:
- [x] Upgrade sigstore/cosign dependency.
- [x] Update code to conform to the new function signatures.
- [x] Check and fix code and tests if needed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,10 +11,10 @@ require (
 	github.com/buildpacks/lifecycle v0.10.2
 	github.com/containerd/containerd v1.5.9 // indirect
 	github.com/ghodss/yaml v1.0.0
-	github.com/go-openapi/spec v0.20.3
+	github.com/go-openapi/spec v0.20.4
 	github.com/google/go-cmp v0.5.7
-	github.com/google/go-containerregistry v0.8.0
-	github.com/google/go-containerregistry/pkg/authn/k8schain v0.0.0-20210610160139-c086c7f16d4e
+	github.com/google/go-containerregistry v0.8.1-0.20220125170349-50dfc2733d10
```
Comment (matthewmcnew) [inline go.mod:16]: Can we use a released version of ggcr?

Comment (stormqueen1990) [reply go.mod:16]: Will the replace directive on line 46 take care of that, perhaps? If I recall correctly the project does not build with this version but it is a transitive dependency of the version bump.

**Review by tomkennedy513 — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -16,46 +16,45 @@ require (
...
-	k8s.io/apimachinery v0.21.3
-	k8s.io/client-go v0.21.3
-	k8s.io/code-generator v0.20.11
-	k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
-	knative.dev/pkg v0.0.0-20210902173607-844a6bc45596
-	sigs.k8s.io/structured-merge-diff v0.0.0-20190525122527-15d366b2352e // indirect
-	sigs.k8s.io/structured-merge-diff/v3 v3.0.0 // indirect
+	k8s.io/api v0.22.5
+	k8s.io/apimachinery v0.22.5
+	k8s.io/client-go v0.22.5
```
Comment (matthewmcnew) [inline go.mod:45]: This bumps the min version of k8s. We should document in the eventual release notes.

**Review by matthewmcnew — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (tylerphelan) [reply go.mod:45]: k8s.io/code-generator v0.20.11 as well

Comment (stormqueen1990) [reply go.mod:45]: Will update!

Comment (stormqueen1990) [reply go.mod:45]: @matthewmcnew @tylerphelan I added new replace directives to lock those versions down, could you please advise if this is enough?

**Review by tylerphelan — APPROVED** [review]

**Pull request merged**